File: app/src/main/java/com/mithril/mobilegoldenleaf/database/dao/AddressDao.java
package com.mithril.mobilegoldenleaf.database.dao;

import androidx.room.Dao;
import androidx.room.Insert;
import androidx.room.Query;
import androidx.room.Update;

import com.mithril.mobilegoldenleaf.models.Address;

import java.util.List;

@Dao
public interface AddressDao {

    @Insert
    long save(Address address);

    @Update
    void edit(Address address);

    @Query("SELECT * FROM Address")
    List<Address> all();

}

File: app/src/main/java/com/mithril/mobilegoldenleaf/persistence/repository/AddressDao.kt
package com.mithril.mobilegoldenleaf.persistence.repository

import androidx.room.Dao
import androidx.room.Insert
import androidx.room.Query
import androidx.room.Update

import com.mithril.mobilegoldenleaf.models.Address

@Dao
interface AddressDao {

    @Insert
    fun save(address: Address): Long

    @Update
    fun edit(address: Address)

    @Query("SELECT * FROM Address")
    fun all(): List<Address>

}
